app/models/manageiq/providers/inventory/persister/builder.rb

```diff
@@ -227,8 +226,14 @@ def auto_model_class
         # Check that safe_constantize returns our expected class_name, if not then
         # return the base class.
         #
-        # safe_constantize can return different similar class ( some Rails auto-magic :/ )
-        provider_class.to_s == class_name ? provider_class : "::#{name.to_s.classify}".safe_constantize
+        # safe_constantize can return different similar class (if it is able to resolve the
+        # class in the hierarchy even though it isn't at the same hierarchy depth we are expecting.
+        return provider_class if provider_class.to_s == class_name
+
+        klass = "::#{name.to_s.classify}".safe_constantize
+        raise NotSubclassedError, "#{name} should be subclassed under #{manager_class}" if klass && klass.sti? && !@options[:without_sti]
```

Comment:
As discussed, I think we should tweak this error message a bit to give the author more info...something like

```suggestion
        raise NotSubclassedError, "expected #{name} to be found subclassed as #{provider_class}, but instead found #{klass}" if klass && klass.sti? && !@options[:without_sti]
```